network/endpoint_windows.go

```diff
@@ -63,36 +118,36 @@ func (nw *network) newEndpointImpl(epInfo *EndpointInfo) (*endpoint, error) {
 		hnsEndpoint.PrefixLength = uint8(pl)
 	}
 
-	// Marshal the request.
-	buffer, err := json.Marshal(hnsEndpoint)
-	if err != nil {
-		return nil, err
-	}
-	hnsRequest := string(buffer)
-
 	// Create the HNS endpoint.
-	log.Printf("[net] HNSEndpointRequest POST request:%+v", hnsRequest)
-	hnsResponse, err := hcsshim.HNSEndpointRequest("POST", "", hnsRequest)
-	log.Printf("[net] HNSEndpointRequest POST response:%+v err:%v.", hnsResponse, err)
+	log.Printf("[net] HostComputeEndpoint CREATE id:%+v", hnsEndpoint)
+	hnsResponse, err := hnsEndpoint.Create()
 	if err != nil {
 		return nil, err
 	}
 
 	defer func() {
 		if err != nil {
-			log.Printf("[net] HNSEndpointRequest DELETE id:%v", hnsResponse.Id)
-			hnsResponse, err := hcsshim.HNSEndpointRequest("DELETE", hnsResponse.Id, "")
-			log.Printf("[net] HNSEndpointRequest DELETE response:%+v err:%v.", hnsResponse, err)
+			delEndpoint(hnsResponse.Id)
 		}
 	}()
 
-	// Attach the endpoint.
-	log.Printf("[net] Attaching endpoint %v to container %v.", hnsResponse.Id, epInfo.ContainerID)
-	err = hcsshim.HotAttachEndpoint(epInfo.ContainerID, hnsResponse.Id)
-	if err != nil {
-		log.Printf("[net] Failed to attach endpoint: %v.", err)
-		return nil, err
-	}
+	/*
+		// TODO: RS5 at release does not process V2 Endpoint Schema, using V1 for now.
+		// Create the endpoint object.
+		gatewayAddr := net.ParseIP(hnsResponse.Routes[0].NextHop)
+		ep := &endpoint{
+			Id:          infraEpName,
+			HnsId:       hnsResponse.Id,
+			SandboxKey:  epInfo.ContainerID,
```

Comment:
what if we want to assign this endpoint to particular network compartment instead of a container? should we extend this part of the code in order to support that?

Reply:
We can add that feature later if it turns out to be desirable. I'd like to separate that additional functionality from this schema update if possible. :)

Reply:
sure i completely agree. i just wanted to verify do i understand this part of the code correctly since we will need that functionality. Ill follow up over email with what are we trying to achieve ;)